subsys/net/ip/dhcpv4.c

```diff
@@ -881,13 +896,18 @@ static bool dhcpv4_parse_options(struct net_pkt *pkt,
 			break;
 		}
 		default:
-			NET_DBG("option unknown: %d", type);
+			if (unhandled) {
+				NET_DBG("option unknown: %d", type);
```

Comment:
I'd rather add `else` here, with `option unknown, handled by callback` or similar.

Reply:
Ok, I'll add the else above and remove the whole if statement here.